oauth2client/flask.py

```diff
@@ -0,0 +1,329 @@
+            console.
+        * OAUTH2_CLIENT_ID the oauth2 credentials' client ID. This is only
+            needed if OAUTH2_CLIENT_SECRETS_JSON is not specified.
+        * OAUTH2_CLIENT_SECRET the oauth2 credentials' client secret. This is
+            only needed if OAUTH2_CLIENT_SECRETS_JSON is not specified.
+    """
+
+    def __init__(self, app=None, *args, **kwargs):
+        self.app = app
+        if app is not None:
+            self.init_app(app, *args, **kwargs)
+
+    def init_app(self, app, scopes=None, client_secrets_file=None,
+                 client_id=None, client_secret=None, authorize_callback=None,
+                 logout_callback=None):
+        self.app = app
+
+        if not scopes:
+            scopes = app.config.get('OAUTH2_SCOPES', DEFAULT_SCOPES)
+
+        self.scopes = scopes
+
+        self.client_id, self.client_secret = client_id, client_secret
+
+        if not self.client_id or not self.client_secret:
+            if client_secrets_file:
+                self.client_id, self.client_secret = self._load_client_secrets(
+                    client_secrets_file)
+            elif 'OAUTH2_CLIENT_SECRETS_JSON' in app.config:
+                self.client_id, self.client_secret = self._load_client_secrets(
+                    app.config['OAUTH2_CLIENT_SECRETS_JSON'])
+            else:
+                self.client_id, self.client_secret = self._load_app_config(app)
+
+        app.register_blueprint(self._create_blueprint())
+
+        self.authorize_callback = authorize_callback
+        self.logout_callback = logout_callback
+
+        app.after_request(self._update_session)
+
+    def _load_client_secrets(self, filename):
+        client_type, client_info = clientsecrets.loadfile(filename)
+        if client_type != clientsecrets.TYPE_WEB:
+            raise ValueError(
+                "The flow specified in %s isn't supported." % client_type)
+
+        return client_info['client_id'], client_info['client_secret']
+
+    def _load_app_config(self, app):
+        if ('OAUTH2_CLIENT_SECRET' not in app.config or
+                'OAUTH2_CLIENT_ID' not in app.config):
+            raise ValueError(
+                "OAuth2 configuration could not be found. Either "
+                "OAUTH2_CLIENT_SECRETS_JSON or OAUTH2_CLIENT_ID and "
+                "OAUTH2_CLIENT_SECRET must be specified.")
+
+        return (app.config['OAUTH2_CLIENT_ID'],
+                app.config['OAUTH2_CLIENT_SECRET'])
+
+    def _update_session(self, resp):
+        """Ensures that the session's copy of the credentials is updated
+        when the credential refreshes its access token."""
+        ctx = _app_ctx_stack.top
+        if hasattr(ctx, 'oauth2_credentials'):
+            session['oauth2credentials'] = ctx.oauth2_credentials.to_json()
+        return resp
+
+    def _request_user_info(self):
+        """Gets the user's profile information from Google+
```

Comment:
As discussed, this might be higher level than we want to go in oauth2client. There is a more comprehensive Google Signin library that should cover this sort of case.